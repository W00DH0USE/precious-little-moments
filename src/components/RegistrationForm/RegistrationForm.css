.RegistrationForm {
  background-color: transparent;
}

label {
  font-size:15px;
  font-weight:700
}

hr {
  border:0;
  border-top:1px solid #ccc;
  margin-bottom:15px
}

.password-requirements {
  font-size: 77%;
  width: 60%;
  margin: 0 auto;
}

@media only screen and (min-width: 768px) {
  .RegistrationForm {
    width: 450px;
    height: 500px;
  }
}