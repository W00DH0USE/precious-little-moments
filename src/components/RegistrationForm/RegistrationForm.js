
import React, { Component } from "react";
import "./RegistrationForm.css";
import { Button, Required, Form, Input2 } from "../Utils/Utils";
import config from '../../config';

const { API_BASE_URL } = config


export default class RegistrationForm extends Component {
   constructor(){
    super()
    this.state = { 
      error: " " 
    };
  } 

  handleUserSubmit = e => {
    e.preventDefault();
    const { first_name, last_name, email, password } = e.target;

    const user = {
      first_name: first_name.value,
      last_name: last_name.value,
      email: email.value,
      password: password.value,
    }

    fetch(`${API_BASE_URL}/users`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(user)
    })
    .then(res => {
      if (res.ok) {
        return res.json()
      } else {
        throw new Error(res.json())
      }
    })
    .then(user => {
      window.location.href = '/login'
    })
    .catch(error => {
      this.setState({error: error})
    })
  }

  render() {
     const { error } = this.state; 
    return (
      <Form className="RegistrationForm" onSubmit={this.handleUserSubmit}>
        <div role="alert">{error && <p className="red">{error}</p>}</div>
        <div className="first_name">
          <h2 className="Form__title">Precious Little Moment: <span><em>sign up</em></span></h2>
          <hr />
          <label htmlFor="first_name">
            First Name <Required />
          </label>
          <Input2
            name="first_name"
            type="text"
            required
            id="first_name"
            autoComplete="off"
          ></Input2>
        </div>
        <div className="last_name">
          <label htmlFor="last_name">
            Last Name <Required />
          </label>
          <Input2
            name="last_name"
            type="text"
            required
            id="last_name"
            autoComplete="off"
          ></Input2>
        </div>
        <div className="user_name">
          <label htmlFor="email">
            Email <Required />
          </label>
          <Input2
            name="email"
            type="email"
            required
            id="email"
            autoComplete="off"
          ></Input2>
        </div>
        <div className="password">
          <label htmlFor="password">
            Password <Required />
          </label>
          <Input2
            name="password"
            type="password"
            required
            id="password"
            autoComplete="off"
          ></Input2>
        </div>
        <Button className="button demo-button" type="submit" style={{ width: "200px" }}>Register</Button>
      </Form>
    );
  }
}
