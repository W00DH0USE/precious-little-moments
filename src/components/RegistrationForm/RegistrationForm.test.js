import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RegistrationForm from './RegistrationForm';


describe(`Registration Form Component`, () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<RegistrationForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});