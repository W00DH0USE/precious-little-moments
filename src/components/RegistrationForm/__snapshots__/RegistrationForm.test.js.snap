// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Form Component renders without crashing 1`] = `
<Form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  >
    <p
      className="red"
    >
       
    </p>
  </div>
  <div
    className="first_name"
  >
    <h2
      className="Form__title"
    >
      Precious Little Moment: 
      <span>
        <em>
          sign up
        </em>
      </span>
    </h2>
    <hr />
    <label
      htmlFor="first_name"
    >
      First Name 
      <Required />
    </label>
    <Input2
      autoComplete="off"
      id="first_name"
      name="first_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="last_name"
  >
    <label
      htmlFor="last_name"
    >
      Last Name 
      <Required />
    </label>
    <Input2
      autoComplete="off"
      id="last_name"
      name="last_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="user_name"
  >
    <label
      htmlFor="email"
    >
      Email 
      <Required />
    </label>
    <Input2
      autoComplete="off"
      id="email"
      name="email"
      required={true}
      type="email"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="password"
    >
      Password 
      <Required />
    </label>
    <Input2
      autoComplete="off"
      id="password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <Button
    className="button demo-button"
    style={
      Object {
        "width": "200px",
      }
    }
    type="submit"
  >
    Register
  </Button>
</Form>
`;
